src/app/shopping-cart/shopping-cart.service.ts

```diff
@@ -15,10 +16,61 @@ export class ShoppingCartService {
 
     getQty(){
         this.cartItemNum = 0;
-        let arr = this.getShoppingCart();
-            arr.forEach((i) => {
-            this.cartItemNum += i.qty;
-        });
+        let reducer = (acc, item) => {
+            return acc + item.qty;
+        }
+
+        this.cartItemNum = this.getShoppingCart().reduce(reducer, this.cartItemNum);
         return this.cartItemNum;
     }
+
+    setToLocalStorage(){
```

Comment:
This does not follow MVC pattern. I would expect something like this"

Where item is an instance of Item:
ShoppingCartService.addItem(item);
ShoppingCartService.setItems(items);
ShoppingCartService.getItems(); 
Notice how I am not exposing how I'm getting these items, the component doesn't care.